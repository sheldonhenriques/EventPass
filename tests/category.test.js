const ejs = require('ejs');
const fs = require('fs');

describe('Category Template', () => {
  test('renders categories correctly', () => {
    const categories = [
      { name: 'Category 1', image: 'category1.jpg' },
      { name: 'Category 2', image: 'category2.jpg' },
    ];

    const template = fs.readFileSync('views/categories/category.ejs', 'utf8');

    const renderedTemplate = ejs.render(template, { categories });
    expect(renderedTemplate).toMatchSnapshot();
  });
});