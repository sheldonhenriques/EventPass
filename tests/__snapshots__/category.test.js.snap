// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Template renders categories correctly 1`] = `
"<div class="container my-4">
    <h1 class="mb-4 text-left"><b>Categories</b></h1>
    <div class="categories-container d-flex overflow-auto">
      
        <div class="category-item flex-shrink-0 mr-3">
          <img src="category1.jpg" alt="Category 1">
          <h3>Category 1</h3>
        </div>
      
        <div class="category-item flex-shrink-0 mr-3">
          <img src="category2.jpg" alt="Category 2">
          <h3>Category 2</h3>
        </div>
      
    </div>
  </div>"
`;
